pkg/history/sqlitestore/store_test.go

```diff
@@ -0,0 +1,470 @@
+package sqlitestore
+
+import (
+	"context"
+	"crypto/rand"
+	"fmt"
+	"os"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zaptest"
+)
+
+func TestDatabase_Insert(t *testing.T) {
+	db := newTestDB(t)
+	ctx := context.Background()
+
+	// Using fixed origin time for test determinism
+	originTime := time.Date(2023, 1, 1, 12, 0, 0, 0, time.UTC)
+	source := "test-source"
+
+	// Insert multiple records sequentially
+	records := []Record{
+		{Source: source, Payload: []byte("test-payload-1"), CreateTime: originTime.Add(-2 * time.Hour)},
+		{Source: source, Payload: []byte("test-payload-2"), CreateTime: originTime.Add(-1 * time.Hour)},
+		{Source: source, Payload: []byte("test-payload-3"), CreateTime: originTime},
+	}
+
+	// Use zero-sized slice with capacity
+	insertedRecords := make([]Record, 0, len(records))
+
+	// Insert all records and collect the results
+	for _, r := range records {
+		record, err := db.Insert(ctx, r)
+		if err != nil {
+			t.Fatalf("unexpected error inserting record: %v", err)
+		}
+
+		if record.ID == 0 {
+			t.Errorf("expected valid record ID, got record.ID=%q", record.ID)
+		}
+
+		if record.Payload == nil || string(record.Payload) != string(r.Payload) {
+			t.Errorf("expected payload to match %q, got %q", r.Payload, record.Payload)
+		}
+
+		insertedRecords = append(insertedRecords, record)
+	}
+
+	// Verify all records were stored correctly
+	verifyRecords(t, db, ctx, records)
+}
+
+// verifyRecords retrieves all records for a source from the database and verifies they match expected records.
+func verifyRecords(t *testing.T, db *Database, ctx context.Context, expected []Record) {
+	t.Helper()
+
+	// Create a slice with capacity to hold all expected records
+	retrievedRecords := make([]Record, len(expected)+1)
+
+	// Fetch all records using zero-value Records to indicate full range
+	count, err := db.Read(ctx, "", 0, 0, false, retrievedRecords)
+	if err != nil {
+		t.Fatalf("unexpected error reading records: %v", err)
+	}
+
+	if count != len(expected) {
+		t.Errorf("expected %d records, got %d", len(expected), count)
+	}
+
+	// Verify each record has correct data
+	for i := range count {
+		if retrievedRecords[i].ID == 0 {
+			t.Errorf("record %d: missing ID", i)
+		}
+
+		diff := cmp.Diff(expected[i], retrievedRecords[i],
+			cmpopts.IgnoreFields(Record{}, "ID"),
+			cmpopts.EquateApproxTime(time.Millisecond),
+		)
+		if diff != "" {
+			t.Errorf("record %d: data mismatch (-want +got):\n%s", i, diff)
+		}
+	}
```

Comment:
You could probably do this entire comparison using `cmp.Diff` which might help with added/deleted comparisons too